docs/aws/audit/esmonitoring/rules/es_https_only.mdx

```diff
@@ -23,6 +23,246 @@ AWSWAF, HITRUST, SOC2, PCIDSS
+</Accordion>
+
+</AccordionGroup>
+</Tab>
+<Tab title='Cause'>
+### Check Cause
+<AccordionGroup>
+<Accordion title='Using Console' defaultOpen='true'>
+1. Sign in to the AWS Management Console.
+2. Navigate to the Amazon Elasticsearch Service (Amazon ES) dashboard.
+3. In the navigation pane, choose "Domains".
+4. In the "Domains" list, select the Elasticsearch domain that you want to examine.
+5. In the "Domain overview" section, check the "Domain endpoint" value. If the protocol is not HTTPS, then the selected Amazon Elasticsearch domain is not using secure communication (HTTPS) for all the traffic to and from the domain.
+</Accordion>
+
+<Accordion title='Using CLI'>
+1. First, you need to install and configure AWS CLI on your local machine. You can download it from the official AWS website and configure it using the "aws configure" command. You will be prompted to provide your AWS Access Key ID, Secret Access Key, Default region name, and Default output format.
+
+2. Once the AWS CLI is set up, you can use the "describe-elasticsearch-domain-config" command to retrieve the configuration information of the specified Elasticsearch domain. The command syntax is as follows:
+
+   ```
+   aws es describe-elasticsearch-domain-config --domain-name <your_domain_name>
+   ```
+
+   Replace `<your_domain_name>` with the name of your Elasticsearch domain.
+
+3. The output of the command will be in JSON format. You need to check the "DomainConfig" section in the output. Under this section, look for the "ElasticsearchClusterConfig" field. This field contains the configuration settings for the Elasticsearch domain.
+
+4. Specifically, you need to check the "TLSSecurityPolicy" field under the "DomainEndpointOptions" section. If the value of this field is not "Policy-Min-TLS-1-2-2019-07", it means that the Elasticsearch domain is not using HTTPS only.
+</Accordion>
+
+<Accordion title='Using Python'>
+1. Import the necessary libraries: You will need the boto3 library for AWS services and the requests library to make HTTP requests.
+
+```python
+import boto3
+import requests
+```
+
+2. Create a session using your AWS credentials and initialize the ElasticSearch client:
+
+```python
+session = boto3.Session(
+    aws_access_key_id='YOUR_ACCESS_KEY',
+    aws_secret_access_key='YOUR_SECRET_KEY',
+    region_name='YOUR_REGION'
+)
+
+es_client = session.client('es')
+```
+
+3. Get the list of all ElasticSearch domains and check the domain configuration:
+
+```python
+response = es_client.list_domain_names()
+
+for domain in response['DomainNames']:
+    domain_name = domain['DomainName']
+    domain_config = es_client.describe_elasticsearch_domain(DomainName=domain_name)
+```
+
+4. Check if HTTPS is enabled for the domain:
+
+```python
+domain_endpoint = domain_config['DomainStatus']['Endpoints']['vpc']
+response = requests.get('http://' + domain_endpoint)
+
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>